### PR TITLE
Used pre-filtered data

### PR DESCRIPTION
**Pull request by jukent**
Filtering the data is no longer part of the workflow, smaller datasets are hosted and the notebook uses them.

### CODE REVIEW

File: Bias-Correction/bias_correction.ipynb [hunk 1]
```diff
@@ -54,28 +35,41 @@
...
    "metadata": {},
    "outputs": [],
    "source": [
-    "hist_file = '../../Seth Comparison/uas.hist.CanESM2.CRCM5-UQAM.day.NAM-44i.raw.nc'\n",
-    "ds_hist = xr.open_dataset(hist_file)"
+    "path = '/glade/u/home/jkent/data/bias_correction_CO'\n",
+    "\n",
+    "hist_file = path + '/uas.hist.CanESM2.CRCM5-UQAM.day.NAM-44i.raw.Colorado.nc'\n",
+    "rcp85_file = path + '/uas.rcp85.CanESM2.CRCM5-UQAM.day.NAM-44i.raw.Colorado.nc'\n",
+    "meas_file = path + '/uas.gridMET.NAM-44i.Colorado.nc'"
```
Comment (andersy005) [inline Bias-Correction/bias_correction.ipynb]: @jukent,  #38 addressed this issue of needing to use different paths depending on whether you are on Cheyenne/Casper or not. 

As a result of #38, theses lines aren't needed. We should be able to add this notebook to the list of notebooks to tests: https://github.com/NCAR/notebook-gallery/blob/97bf3a32e1d55978d37c562747b56c3aa876c7fa/.circleci/config.yml#L35

**Comment by bonnland** [comment]
@jukent When you think the notebook is ready for testing, could you update the file .circleci/config.yml to run your notebook as part of the tests?   Something like changing this line:

`pytest --junitxml=test-reports/junit.xml -v -s --current-env --nbval-lax mpas/*.ipynb`

to 

`pytest --junitxml=test-reports/junit.xml -v -s --current-env --nbval-lax mpas/*.ipynb Bias-correction/*.ipynb`

**Review by bonnland — COMMENTED** [review]
Without the typos in the folder name, of course...  :)

**Comment by jukent** [comment]
Just changed that line in config.yml

**Comment by bonnland** [comment]
After the tests pass, I will approve.  Thanks Julia!

**Comment by jukent** [comment]
*fingers crossed*

**Comment by bonnland** [comment]
It looks like leading slashes in the file name string are preventing data file download; exact match is probably required.   That's my reading of the log file from CircleCI.

**Comment by jukent** [comment]
It is failing after the call to the ngallery_utils.DATASETS.fetch:

"File '/uas.hist.CanESM2.CRCM5-UQAM.day.NAM-44i.raw.Colorado.nc' is not in the registry."

**Comment by jukent** [comment]
Just removed the slashes, hopefully we'll have better luck this time.

File: Bias-Correction/bias_correction.ipynb [hunk 2]
```diff
@@ -54,28 +35,39 @@
...
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    "# Or if not on Cheyenne\n",
+    "hist_file = DATASERTS.fetch(\"/uas.hist.CanESM2.CRCM5-UQAM.day.NAM-44i.raw.Colorado.nc\")\n",
+    "rcp85_file = DATASERTS.fetch(\"/uas.rcp85.CanESM2.CRCM5-UQAM.day.NAM-44i.raw.Colorado.nc\")\n",
+    "meas_file = DATASETS.fetch(\"uas.gridMET.NAM-44i.Colorado.nc\")"
```
Comment (andersy005) [inline Bias-Correction/bias_correction.ipynb]: Can we remove this cell from the notebook?

**Comment by jukent** [comment]
We got further, but it timed out in cell 14.

**Comment by bonnland** [comment]
It's possible that the timeout value is a parameter for pytest; @jukent could you check the documentation for pytest to see?   I'm going to also check whether we can use multiple workers to run the tests.

**Comment by andersy005** [comment]
> It's possible that the timeout value is a parameter for pytest;

This is part of CircleCI settings.  If it has been 10 minutes since the last output, CircleCI cancels and stops the build. See: https://support.circleci.com/hc/en-us/articles/360007188574-Build-has-hit-timeout-limit

**Comment by bonnland** [comment]
Good catch; luckily we can override the default value.   Good catch, Anderson!

File: Bias-Correction/bias_correction.ipynb (hunk 2 above)
Comment (jukent) [reply Bias-Correction/bias_correction.ipynb]: removed

**Comment by jukent** [comment]
Are markdown cells counted when it says it timed out at cell 14? I think it is timing out when calculating the future statistics (`_calc_fut_stats`) but it could be earlier if markdown is counted.

**Comment by jukent** [comment]
Would I go to 'Rerun job with SSH' and enter:

- run:  
    command: . ./bias_correction.ipynb
    no_output_timeout: 30m

Or do I enter it somewhere else. I think I have to wait for these tests to complete before I can rerun it in SSH.

File: .circleci/config.yml [hunk 3]
```diff
@@ -32,7 +32,7 @@ jobs:
           name: Run Tests
           command: |
             source activate ${ENV_NAME}
-            pytest --junitxml=test-reports/junit.xml -v -s --current-env --nbval-lax mpas/*.ipynb
+            pytest --junitxml=test-reports/junit.xml -v -s --current-env --nbval-lax mpas/*.ipynb Bias-Correction/*.ipynb
```
Comment (andersy005) [inline .circleci/config.yml]: Adding `no_output_timeout` here may address the issue 


```yaml
command: |
    no_output_timeout: 30m
    source activate ${ENV_NAME}	            
```

Comment (jukent) [reply .circleci/config.yml]: `no_output_timeout` has to go at the same indexing as the `command:` Trying it as:

```
- run:
          name: Run Tests
          command: |  
                 source activate ${ENV_NAME}
                 pytest --junitxml=test-reports/junit.xml -v -s --current-env ...
          no_output_timeout: 30m
```

**Comment by jukent** [comment]
"Too long with no output (exceeded 30m0s): context deadline exceeded"

I don't want to keep adding time to the tests. I am reading in to how to get more information out of CircleCi (as in confirmation of what cell 14 is).

**Comment by bonnland** [comment]
You can try removing the "-s" flag from the pytest line; it's possible this will increase the output.

**Comment by jukent** [comment]
I see this in the config.yml file:
```
# Tell Circle to store the documentation output in a folder that we can access later
      - store_test_results:
          path: test-reports
```
but I am not sure where the test-reports folder now is.

**Comment by jukent** [comment]
@bonnland I split the cell that I suspected and we have now made it to cell 15 before timing out, so that's progress. If it still times out I will make the change @andersy005 suggested.

**Comment by bonnland** [comment]
@jukent If you're logged into CircleCI using your github auth (I was not by just clicking the link), then you will see an "Artifacts" Tab on the CircleCI Job page.   But the artifact right now is just a repeat of what you see in the test output.   The test report is considered an Artifact.

**Comment by bonnland** [comment]
The CircleCI Jobs page says that the tests are running on a machine with 2 CPUs and 4 GB memory.   Our laptops have 16GB of memory.   It's possible that the notebook is running low on memory, which would potentially cause thrashing (slowdown).

**Comment by bonnland** [comment]
@jukent Rather than republish the data files, is it possible to select a smaller bounding box in the notebook and then pass the data subset to these longer calculations?   It might be considered educational to do this, as data subsetting is a common need for scientists.

**Comment by bonnland** [comment]
Another possibility is to delete variables in the notebook that are no longer needed prior to running the problematic cell, especially if the variables take up a lot of memory.     There are two ways to delete variables, as described here:

https://teamtreehouse.com/community/how-to-delete-a-variable-in-python

**Comment by jukent** [comment]
```
fut_std_bc = fut_std * scale
```
Runs fine but
```
fut_mean_bc = fut_mean + shift
```
stalls out. What could be the important difference between these two lines? I will try subsetting the data/deleting old variables.

If it is a memory thing that could be fixed by deleting variables, I bet that if I swapped which line came first calculating fut_mean_bc would run but fut_std_bc would time out.

**Comment by jukent** [comment]
@bonnland I limited the data within the notebook and voila! Thanks for the tips and motivation :)

**Comment by bonnland** [comment]
And, yet another possibility is to put expensive commands inside functions.   Because python automatically releases memory used in a function when the function exits.   But good job, and way to stick with it Julia!

**Comment by bonnland** [comment]
@jukent Is there any more code refactoring or cleanup you want in the notebook before merge happens?

**Comment by jukent** [comment]
I just took some of the cells I had split up to hopefully help with the time-out issue and recombined them. After that passes we're good to go.

**Review by bonnland — COMMENTED** [review]
Everything looks fine, except I wonder if the cell numbers can be reset somehow?  It looks like they start with 46 right now.   Not a big deal, just thought it would make better sense if it was easy to fix.

**Review by bonnland — APPROVED** [review]
Looks good!

**Comment by jukent** [comment]
> Looks good!

Thanks! Cell numbers should be fixed now. I had to trick VS into thinking there was a different change.

**Pull request merged**